### PR TITLE
Rename cmake ie_ macros and ie_parallel script

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/cmake/ov_parallel.cmake [hunk 1]
```diff
@@ -398,4 +398,8 @@ function(set_ie_threading_interface_for TARGET_NAME)
         find_package(Threads REQUIRED)
         ie_target_link_libraries(${TARGET_NAME} ${LINK_TYPE} Threads::Threads)
     endif()
+endfunction(set_ov_threading_interface_for)
+
+function(set_ie_threading_interface_for TARGET_NAME)
```
Comment (ilya-lavrenov) [inline src/cmake/ov_parallel.cmake:403]: let's insert a warning that new macro / function should be used

File: src/cmake/ov_parallel.cmake [hunk 2]
```diff
@@ -242,7 +242,7 @@ macro(ov_find_package_tbb)
     endif()
 endmacro()
 
-function(set_ie_threading_interface_for TARGET_NAME)
+function(set_ov_threading_interface_for TARGET_NAME)
```
Comment (ilya-lavrenov) [inline src/cmake/ov_parallel.cmake:245]: maybe we need to give a better name like `ov_set_threading_interface_for`

Comment (ilyachur) [reply src/cmake/ov_parallel.cmake]: Fixed

File: src/cmake/ov_parallel.cmake (hunk 1 above)
Comment (ilyachur) [reply src/cmake/ov_parallel.cmake:403]: Added

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
Looks like we need to fix oneDNN after merging this PR:
```cmake
CMake Warning at cmake/developer_package/IEDevScriptsConfig.cmake:279 (message):
  This function is deprecated.  Please use ov_mark_target_as_cc(TARGET_NAME)
  instead.
Call Stack (most recent call first):
  src/plugins/intel_cpu/thirdparty/onednn/src/CMakeLists.txt:116 (ie_mark_target_as_cc)
  src/plugins/intel_cpu/thirdparty/onednn/src/common/CMakeLists.txt:55 (enable_conditional_compilation4)


CMake Warning at cmake/developer_package/IEDevScriptsConfig.cmake:279 (message):
  This function is deprecated.  Please use ov_mark_target_as_cc(TARGET_NAME)
  instead.
Call Stack (most recent call first):
  src/plugins/intel_cpu/thirdparty/onednn/src/CMakeLists.txt:116 (ie_mark_target_as_cc)
  src/plugins/intel_cpu/thirdparty/onednn/src/cpu/CMakeLists.txt:125 (enable_conditional_compilation4)


CMake Warning at cmake/developer_package/IEDevScriptsConfig.cmake:279 (message):
  This function is deprecated.  Please use ov_mark_target_as_cc(TARGET_NAME)
  instead.
Call Stack (most recent call first):
  src/plugins/intel_cpu/thirdparty/onednn/src/CMakeLists.txt:116 (ie_mark_target_as_cc)
  src/plugins/intel_cpu/thirdparty/onednn/src/cpu/x64/CMakeLists.txt:63 (enable_conditional_compilation4)
```

**Pull request merged**